compiler/il/Aliases.cpp

```diff
@@ -329,7 +333,8 @@ OMR::SymbolReference::getUseDefAliasesBV(bool isDirectCall, bool includeGCSafePo
              symRefTab->isNonHelper(self(), TR::SymbolReferenceTable::osrFearPointHelperSymbol) ||
              symRefTab->isNonHelper(self(), TR::SymbolReferenceTable::potentialOSRPointHelperSymbol) ||
              symRefTab->isNonHelper(self(), TR::SymbolReferenceTable::eaEscapeHelperSymbol) ||
-             symRefTab->isNonHelper(self(), TR::SymbolReferenceTable::objectEqualityComparisonSymbol))
+             symRefTab->isNonHelper(self(), TR::SymbolReferenceTable::objectEqualityComparisonSymbol) ||
+             symRefTab->isNonHelper(self(), TR::SymbolReferenceTable::nonNullableArrayNullStoreCheckSymbol))
             {
```

Comment:
This seems to make the aliasing conservative (but it is obviously functionally ok). Is this the final state you envision or were you thinking of just delivering the basic support now for this new symbol and look at aliasing refinements for all these helper calls later ?

Reply:
Thanks for posing this question, Vijay @vijaysun-omr.  I was actually just following along with the implementations of other non-helper symbols, such as `objectEqualityComparisonSymbol`, and I wasn't thinking too much about refining the aliasing.  We could certainly do better in terms of the aliasing. . . .  I can open a follow on issue for that as a potential improvement, if you're OK with this for now.

Reply:
Sure, I think its okay to deal with alias refinement separately. I think we had discussed it in some other context as being one of the optimizations to maybe look at the next stage of building out the valhalla support and so the follow on issue is something to track and fix in the not distant future.